EC-84: Restructure eZ Commerce doc, part 2 (#2)
ezsystems/commerce-documentation: docs/guide/breadcrumbs/breadcrumbs_api/postsilvermodulebreadcrumbsgenerator.md

# PostSilverModuleBreadcrumbsGenerator

`Silversolutions\Bundle\EshopBundle\Service\Breadcrumbs\PostSilverModuleBreadcrumbsGenerator`

> is for requests after a silver.module, which process the controller of the previous silver.module.

Reviewer: "is for" -> "handles"?

> is for requests after a silver.module, which process the controller of the previous silver.module.

Reviewer 2: ```suggestion
handles requests for a `silver.module`, which process the controller of the previous `silver.module`.
```

> **Example**: A silver.module Content item specifies the controller `SisoCustomerCenterBundle:Form:form` in order to display a form. After this form is submitted, it will not request the URL of the silver.module, but the URL of the controller, defined by its route. But nonetheless, the breadcrumbs of the former silver.module shall be displayed. 

Reviewer: Specifies - does it mean "orders ... to do something" / "requests that ... does something"? 
Remove "in order"
", defined" -> "that is defined"
"nonetheless..." -> ", even though eZ Platform still displays breadcrumbs for the silver.module."?

> This generator is responsible for this special case of requests. It listens to two kernel events in order to add some request attributes and save the last used silver.module ID temporarily in the session.

Reviewer: ```suggestion
The generator handles such requests: It listens to two kernel events to add request attributes and temporarily save the ID of the last silver.module in the session.
```

!!! tip

>     silver.modules themselves are standard Content items and their breadcrumbs are handled solely by the parent of this class which is [`EzContentBreadcrumbsGenerator`](ezcontentbreadcrumbsgenerator.md).

Reviewer 2: ```suggestion
    silver.modules themselves are standard Content items and their breadcrumbs are handled solely by the parent of the [`EzContentBreadcrumbsGenerator` class](ezcontentbreadcrumbsgenerator.md).
```

## When it is used

The generator triggers if the session attribute 'silverModuleLocation' is set. This is set by an event listener callback, which is described below.

## Rendering process

> In order to memorize a formerly displayed silver.module element, its ID is stored in the session and restored if the next page matches the controller, which is defined in the respective silver.module. This is achieved via two kernel event listeners.

Reviewer: Who/what momerizes the silver.module? I'd rewriite the sentence in active voice. Also, I'd break it in a series of shorter sentences. I would spell out the object hidden behind "It".

### kernel.controller event

> Method `Silversolutions\Bundle\EshopBundle\Service\Breadcrumbs\PostSilverModuleBreadcrumbsGenerator::onKernelRequest()`

Reviewer: See comment below

> Method `Silversolutions\Bundle\EshopBundle\Service\Breadcrumbs\PostSilverModuleBreadcrumbsGenerator::onKernelRequest()`

Reviewer 2: ```suggestion
`Silversolutions\Bundle\EshopBundle\Service\Breadcrumbs\PostSilverModuleBreadcrumbsGenerator::onKernelRequest()` method
```

> is the listener method for the preparation of the request for a post silver.module controller.

Reviewer 2: ```suggestion
is the listener method that prepares the request for a post silver.module controller.
```

It checks the session for a formerly stored silver.module ID and whether the Content item defines the controller that matched in the current request. If the controller doesn't match, the current request is just a call of an unrelated page and can be ignored. If it does match, the request attributes `silverModule` and `silverModuleLocation` are added for the rendering of the breadcrumbs.

### kernel.response event

> Method `Silversolutions\Bundle\EshopBundle\Service\Breadcrumbs\PostSilverModuleBreadcrumbsGenerator::onKernelResponse()`

Reviewer 2: ```suggestion
`Silversolutions\Bundle\EshopBundle\Service\Breadcrumbs\PostSilverModuleBreadcrumbsGenerator::onKernelResponse()` method
```

> is the listener method the temporary storage of the silver.module ID in the session.

Reviewer: "...for the temporary..."

> is the listener method the temporary storage of the silver.module ID in the session.

Reviewer 2: ```suggestion
is the listener method that temporary stores the silver.module ID in the session.
```
